sys/include/uri_parser.h

```diff
@@ -39,18 +39,43 @@ extern "C" {
  * @brief container that holds all results
  */
 typedef struct {
-    char *scheme;                 /**< scheme */
-    char *userinfo;               /**< userinfo */
-    char *host;                   /**< host */
-    char *port;                   /**< port */
-    char *path;                   /**< path */
-    char *query;                  /**< query */
-    uint16_t scheme_len;          /**< length of @p scheme */
-    uint16_t userinfo_len;        /**< length of @p userinfo */
-    uint16_t host_len;            /**< length of @p host */
-    uint16_t port_len;            /**< length of @p port */
-    uint16_t path_len;            /**< length of @p path */
-    uint16_t query_len;           /**< length of @p query */
+    char *scheme;                   /**< scheme */
+    char *userinfo;                 /**< userinfo */
+
+    /**
+     * @brief host part
+     *
+     * @note for IPv6 addresses, @ref host also includes the brackets
+     * '[' and ']' as well as the interface (with leading '%'), if
+     * present.
+     */
+    char *host;
+
+    /**
+     * @brief @p host without @p interface
+     *
+     * @note for IPv6 addresses, @ref host_no_interface does not
+     * include the brackets '[' and ']'.
+     */
+    char *host_no_interface;
+
+    /**
+     * @brief interface part if @p host is IPv6 address, NULL
+     * otherwise
+     */
+    char *interface;
+
+    char *port;                     /**< port */
+    char *path;                     /**< path */
+    char *query;                    /**< query */
+    uint16_t scheme_len;            /**< length of @p scheme */
+    uint16_t userinfo_len;          /**< length of @p userinfo */
+    uint16_t host_len;              /**< length of @p host */
+    uint16_t host_no_interface_len; /**< length of @p host_no_interface */
+    uint16_t interface_len;         /**< length of @p interface */
+    uint16_t port_len;              /**< length of @p port */
+    uint16_t path_len;              /**< length of @p path */
+    uint16_t query_len;             /**< length of @p query */
```

Comment:
Unrelated, since this is already like this in master, but isn't `@p` for _p_arameters? I usually do something like this

```C
    uint16_t scheme_len;            /**< length of uri_parser_result_t::scheme */
    /* ... */
```

This should also create the correct doc in doxygen.

Reply:
You are right :/ I could also use @ref instead of using the full struct name. Would that be okay?

Reply:
I don't think `s/@p/@ref/` would work here.

Reply:
with `@p`:
![image](https://user-images.githubusercontent.com/739129/97591849-526b0680-1a00-11eb-8c58-b308ea0e2374.png)

with `@ref`:
![image](https://user-images.githubusercontent.com/739129/97592011-7c242d80-1a00-11eb-8e95-864e15463fb7.png)

`@ref` is able to deduce the correct struct, if it is used within the struct documentation.

The latest commit could be a proposal?

Reply:
Confirmed [here](https://4528-7370241-gh.circle-artifacts.com/0/doc/structuri__parser__result__t.html) that these new links are working :-).